design/features/file-upload/README.md

```diff
@@ -0,0 +1,353 @@
+
+### Check your answers with file upload
+
+![Check your answers before submitting the form. Screenshot](screenshots-released-version/007-check-your-answers-file-upload.png)
+
+*Page titled: “Check your answers before submitting your form”*
+
+Once the user has completed filling in the form they land on “Check your answers before submitting the form”. If the question asking for a file was simple, without any guidance, we display the question text in the summary list key. Followed by the name and format of the file in the summary list value and a ‘Change’ action link. The change link has a hidden text that contains the content of the question, so it reads like “Change Upload a photo of your pet”.
+
+### Check your answer page with page heading for file upload question with guidance
+![Check your answers - file upload question text and page heading above it in the summary list](screenshots-released-version/008-check-your-answers-file-upload-page-heading.png)
+
+*Page titled: “Check your answers before submitting your form”*
+
+When a file upload question has a guidance it changes the structure of the question. This means that the main context of the question is likely to be in the page heading and the question text becomes more generic, like ‘Upload a photo”.
+
+If there are multiple file upload questions with a generic question text, then on Check your answers page the form fillers might not be able to review which files they uploaded relate to a question without the context of the page heading.
+We decided that for file upload questions, if they contain a page heading for the guidance we will display the page heading above the question text on the check your answer page to provide more context and help users review their answers before submitting the form. 
+
+
+### Submitted form responses in an email to a form processor
+![Email with form responses to a form processor that includes four attachments. Screenshot](screenshots-released-version/009-email-submitted-form-responses-with-attachments.png)
+
+*Email mockup containing the information submitted in the form with attachments.*
+
+When a form is submitted the responses get sent to an email to processing teams. The email would contain the question answers in the body of an email in a structured way.
+Answers to file upload questions are displayed under a questions text in the format of the uploaded file name and type. And each has appended content saying "(attached to this email)". To make it clear that the answer is an attachment and to help form processors match up the right uploaded file to the right question.
+
+## Measuring
+
+The first release of file upload for [GOV.UK](http://GOV.UK) Forms is mostly essential functionality with limitations, and we know there is more we could do to make it better for form creators and form fillers. We have implemented ways to measure the use of the feature, so that we can understand and prioritise any problematic parts to address. We check for:
```

Comment:
```suggestion
The first release of file upload for GOV\.UK Forms is mostly essential functionality with limitations, and we know there is more we could do to make it better for form creators and form fillers. We have implemented ways to measure the use of the feature, so that we can understand and prioritise any problematic parts to address. We check for:
```